### ISSUE TITLE
WAN2.2 T2V LoRA training - I can't seem to get eval images to work

### ISSUE THREAD

**Issue by kevinpawsey** [tdrussell/diffusion-pipe#346]
This is maybe a training issue (me being trained that is), but I have configured the toml files with the options for creating training images, but it doesn't seem to be working.

in the MainConfig.toml file:
```
eval_datasets = [
  { name = 'eval_images_v1.0', config = '/home/xxxxx/xxxxx/diffusion-pipe/examples/eval_dataset.toml' },
]

# eval settings
eval_every_n_epochs = 1
eval_before_first_step = true
eval_micro_batch_size_per_gpu = 1
eval_gradient_accumulation_steps = 1
save_eval_images = true

[model]
type = 'wan'
ckpt_path = '/home/voxta/diffusion-pipe/models/wan2.2-14b-t2v'
transformer_path = '/home/voxta/diffusion-pipe/models/wan2.2-14b-t2v/low_noise_model'
llm_path = '/home/voxta/diffusion-pipe/models/wan2.2-14b-t2v/models_t5_umt5-xxl-enc-bf16.pth'
dtype = 'bfloat16'
transformer_dtype = 'float8'
min_t = 0
max_t = 0.875
timestep_sample_method = 'logit_normal'
```

(there is other config, but the training itself seems to be working ok, so thought I would just post the eval parts)

then the eval_dataset.toml file:
```
resolutions = [768]
enable_ar_bucket = true
min_ar = 0.5
max_ar = 2.0
num_ar_buckets = 3
frame_buckets = [1]

[[directory]]
path = '/home/xxxxxx/xxxxxx/diffusion-pipe/eval_images'
num_repeats = 1
```

in the eval_images path I have 4 png images with 4 txt caption files named the same (stole them out the main training pool).  I can see that the image latents are being cached in the logs, and I can't see any errors in the logs.
```
2025-08-03 16:40:38
[2025-08-03 17:40:38,808] [INFO] [engine.py:165:__init__] RANK=0 STAGE=0 LAYERS=42 [0, 42) STAGE_PARAMS=153354240 (153.354M) TOTAL_PARAMS=153354240 (153.354M) UNIQUE_PARAMS=153354240 (153.354M)
2025-08-03 16:40:41
Running eval
2025-08-03 16:40:41
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 36/36 [02:10<00:00,  3.64s/it]
....
2025-08-03 17:06:20
Started new epoch: 2
2025-08-03 17:06:23
Running eval
2025-08-03 17:06:23
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 36/36 [02:10<00:00,  3.63s/it]
```

... am I missing something, or do eval images/videos not work with t2v training.

Sorry if it is blatently obvious, but I couldn't get them working when I was training Wan2.1 LoRAs either... but I thought that they would be so helpful to see the progress that I would like them to work (if it is possible!).

TYIA for any help/advice that you can give.

**Comment by tdrussell** [comment]
>creating training images

I think you are expecting it to generate images during training. The validation set is only for measuring stabilized validation loss. You can see it in Tensorboard. And if you took the validation images from the training set, it's not validation loss, but rather an estimate of stabilized training loss. Just keep that in mind.

**Comment by kevinpawsey** [comment]
Hi @tdrussell Thank you for taking time out to reply to my question, that now makes sense.

Is there a source of information so that I can understand how is best to use the eval dataset? Should I be using a copy some samples of what is in the full dataset?

I could indeed see the eval dataset in tensorboard (and in mandb dashboard too). I could see the loss reduce/level out, would this be a fair representation of the learning of the model, or is there a better way?

**Comment by ghost** [comment]
> > creating training images
> 
> I think you are expecting it to generate images during training. The validation set is only for measuring stabilized validation loss. You can see it in Tensorboard. And if you took the validation images from the training set, it's not validation loss, but rather an estimate of stabilized training loss. Just keep that in mind.

Would doing an eval set for i2v wan 2.2 5b training help estimate how well the lora is actually training? Something like validation loss sounds like it could be another measuring stick to guestimate how well a model is converging.
